File: Prueba y error/ejercicios de semana santa php/One php/Library/EmpleadoTiempoCompleto.php
<?php
namespace Library;

class EmpleadoTiempoCompleto {
    private string $nombre;
    private float $sueldoBase;
    private float $subsidio;
    private int $riesgo;

    public function __construct(string $nombre, float $sueldoBase, float $subsidio, int $riesgo) {
        $this->nombre = $nombre;
        $this->sueldoBase = $sueldoBase;
        $this->subsidio = $subsidio;
        $this->riesgo = $riesgo;
    }

    public function getNombre(): string {
        return $this->nombre;
    }

    public function getSueldoBase(): float {
        return $this->sueldoBase;
    }

    public function getSubsidio(): float {
        return $this->subsidio;
    }

    public function getRiesgo(): int {
        return $this->riesgo;
    }

    public function calcularPago(float $totalExtras, float $totalDescuentos): float {
        return $this->sueldoBase + $this->subsidio + $totalExtras - $totalDescuentos;
    }
}

File: Prueba y error/ejercicios de semana santa java/one java/Library/EmpleadoTiempoCompleto.java
package Library;

public class EmpleadoTiempoCompleto extends Empleado {
    public EmpleadoTiempoCompleto(String nombre, double sueldoBase, double subsidio, int riesgo) {
        super(nombre, sueldoBase, subsidio, riesgo);
    }

    @Override
    public double calcularPago(double horasExtras, double totalDescuentos) {
        return super.calcularPago(horasExtras, totalDescuentos);
    }
}
